docs/MultiStepDonationForm-and-FormControllers.md

```diff
@@ -0,0 +1,88 @@
+
+## `SubmitForm`
+
+This is a hidden form. It contains the values from the `FormModel` as hidden fields and is submitted by the `MultiStepDonationForm` when `FormController` invokes the `onSubmit` callback. This data is the POSTed to the Fundraising Application as a standard HTTP POST request. 
+
+## In Summary
+
+* The `MultistepDonationForm` is an event bus.
+* The `FormModel` contains the form state values.
+* The `FormController` handles the main form submission logic.
+* The `FormSteps` handle their own validation and fire submit/next/previous events.
+* Some `FormSteps` modify the model state directly some pass that data to the `FormController` for extra computing.
+* The `SubmitForm` has hidden form fields with `FormModel` values and POSTs to the Fundraising Application. 
+
+```mermaid
+stateDiagram-v2
+    FormModel: Globally available FormModel
+    FormController --> MultiStepDonationForm
+    MultiStepDonationForm --> FormStepOne
+    MultiStepDonationForm --> FormStepTwo
+    MultiStepDonationForm --> FormStepThree
+    MultiStepDonationForm --> SubmitForm: Used to post data to Fundraising Application
+```
+
+Below is an example happy path user flow.
+
+```mermaid
+sequenceDiagram
+    participant FormController
+    participant MultiStepDonationForm
+    participant FormStep1
+    participant FormStep2
+    participant SubmitForm
+    
+    MultiStepDonationForm -->> FormController: add callback to onNext
+    MultiStepDonationForm -->> FormController: add callback to onPrevious
+    MultiStepDonationForm -->> FormController: add callback to onGoToStep
+    MultiStepDonationForm -->> FormController: add callback to onSubmit
+    
+    FormStep1 ->> MultiStepDonationForm: user submits step 1
+    MultiStepDonationForm ->> FormController: pass submit to controller
+    FormController ->> MultiStepDonationForm: calls onNext callback
+    MultiStepDonationForm ->> FormStep2: shows next step
+    
+    FormStep2 ->> MultiStepDonationForm: user submits step 2
+    MultiStepDonationForm ->> FormController: pass submit to controller
+    FormController ->> MultiStepDonationForm: calls onSubmit callback
+    MultiStepDonationForm ->> SubmitForm: posts form
+```
+
+## Notes for the future
+
+This module isn't perfect. Depending on the direction of the future campaign tests we should improve some things about it.
+
+* The form user flows are badly designed. For example, we don't know the reasoning why some forms clear their own state on back. This needs to be clarified before a redesign happens.
+* `FormControllers` These are where we put the smelly code. Event handling is done per sub-form using switch statements to branch the logic. This could be improved somewhat.
```

Comment:
```suggestion
* `FormControllers` These are where we put the smelly code. Event handling is done per sub-form using switch statements to branch the logic. This could be improved somewhat.
* Some of the FormControllers are very similar. We might extract them into shared classes for common sets of form steps.
```